pkg/localstore/gc.go

```diff
@@ -218,6 +227,11 @@ func (db *DB) collectGarbage() (collectedCount uint64, done bool, err error) {
 		db.metrics.GCErrorCounter.Inc()
 		return 0, false, err
 	}
+
+	for _, loc := range locations {
```

Comment:
Instead of passing a context here, you can create a context that is canceled when localstore.Close is called (which could close some new channel to signal it's called). Which would stop the garbage collection when the user calls a close.